### PR TITLE
Fix MiniMaxM2 partial RoPE: map legacy `rotary_dim` to `partial_rotary_factor`

### LINKED ISSUE #48241: MiniMaxM2 silently applies full-head RoPE: config drops the checkpoints' legacy `rotary_dim` field
### System Info

Verified on current `main` (`3d6c720188`, `5.16.0.dev0`); also present in v5.12.1. torch 2.10, linux.

### Reproduction

All released `MiniMaxAI/MiniMax-M2*` checkpoints are trained with **partial RoPE** (`rotary_dim: 64` of `head_dim: 128`) and express it in `config.json` through a legacy `rotary_dim` field — their original `trust_remote_code` config class derived `partial_rotary_factor = rotary_dim / head_dim` in `__init__`. The in-tree `MiniMaxM2Config` keeps `rotary_dim` as an opaque extra attribute and `RopeParametersMixin.convert_rope_params_to_dict` only recognizes the standard `partial_rotary_factor` spelling, so `rope_parameters` never receives a partial factor and `MiniMaxM2RotaryEmbedding` silently rotates **all 128 head dims with a full-width frequency ladder** — both the frequency spacing and the rotate-half pairing are wrong in every attention layer, with no warning.

Config-level repro (no weights needed):

```python
from transformers import MiniMaxM2Config
from transformers.models.minimax_m2.modeling_minimax_m2 import MiniMaxM2RotaryEmbedding

# Mirrors MiniMaxAI/MiniMax-M2* config.json (rotary_dim: 64, head_dim: 128).
config = MiniMaxM2Config(rotary_dim=64)
print(config.rope_parameters)   # {'rope_theta': 5000000.0, 'rope_type': 'default'}  <- no partial factor
print(MiniMaxM2RotaryEmbedding(config).inv_freq.shape)
# torch.Size([64]) -> full-head rotation. Expected torch.Size([32]) for rotary_dim=64.
```

Or directly from the released checkpoint's metadata:

```python
from transformers import AutoConfig
config = AutoConfig.from_pretrained("MiniMaxAI/MiniMax-M2")
print(config.rotary_dim, config.head_dim, config.rope_parameters)
# 64 128 {'rope_theta': 5000000, 'rope_type': 'default'}
```

### Measured impact

We hit this in NVIDIA NeMo AutoModel's checkpoint-parity CI while using the in-tree implementation as the vanilla reference for `MiniMaxAI/MiniMax-M2.7`:

- Full-model logits vs a faithful implementation of the checkpoint's own modeling semantics on a 2,048-token document: **mean KL 13.1, p95 22.4, cosine 0.02** — the model is effectively destroyed while loading cleanly.
- Minimal causal proof: on a tiny randomly-initialized 2-layer model in fp32, the in-tree forward diverges from the reference (cosine 0.885); adding `"partial_rotary_factor": 0.5` to `rope_parameters` on the same config makes the *unmodified* in-tree model match the reference to max-abs 7e-7.

### Expected behavior

`MiniMaxM2Config(rotary_dim=64)` should yield `rope_parameters["partial_rotary_factor"] == 0.5` so the model rotates only the trained 64 dims — matching the checkpoints' original remote-code config. Fix (following the `GPTNeoXConfig` `rotary_pct` precedent): map the legacy field in a `convert_rope_params_to_dict` override. PR incoming.

### PR DESCRIPTION
**Pull request by yuhezhang-ai**
[CPU CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48242) [GPU run-slow](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48242)

# What does this PR do?

Fixes #48241.

All released `MiniMaxAI/MiniMax-M2*` checkpoints are trained with partial RoPE (`rotary_dim: 64` of `head_dim: 128`) and express it through a legacy `rotary_dim` config field; their original remote-code config derived `partial_rotary_factor = rotary_dim / head_dim` in `__init__`. The in-tree `MiniMaxM2Config` keeps `rotary_dim` as an opaque extra attribute, so `rope_parameters` never receives a partial factor and `MiniMaxM2RotaryEmbedding` silently rotates the full head dimension with a full-width frequency ladder — wrong frequency spacing *and* wrong rotate-half pairing in every attention layer (measured: mean KL 13.1 / cosine 0.02 vs a faithful reference on MiniMax-M2.7 over 2,048 tokens; details in the issue).

This maps the legacy field in a `convert_rope_params_to_dict` override on `MiniMaxM2Config` (in `modular_minimax_m2.py`, generated file regenerated), following the `GPTNeoXConfig` `rotary_pct` precedent:

- `MiniMaxM2Config(rotary_dim=64)` → `rope_parameters["partial_rotary_factor"] == 0.5`, `inv_freq` length 32;
- an explicit `partial_rotary_factor` in `rope_parameters` still wins;
- configs without `rotary_dim` are unaffected.

## Before submitting
- [ ] This PR fixes a typo or improves the docs
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request)?
- [x] Was this discussed/approved via a Github issue? #48241
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? `test_legacy_rotary_dim_maps_to_partial_rotary_factor` (covers legacy mapping, explicit-factor priority, and the no-legacy-field case; `utils/check_modular_conversion.py` passes)

## Who can review?

@ArthurZucker @Cyrilvallez

### CODE REVIEW

**Comment by Rocketknight1** [comment]
cc @arthurzucker I think this one is real, and it probably breaks most existing Minimax-M2 checkpoints! The fix is probably valid but deferring to you on where exactly we want it in the code

**Comment by Cyrilvallez** [comment]
Closing as https://github.com/huggingface/transformers/pull/48486 was merged!
